guides/micronaut-microservices-services-discover-eureka/bookinventory/java/src/main/java/example/micronaut/BookInventory.java

```diff
@@ -1,26 +1,24 @@
 package example.micronaut;
 
+import io.micronaut.core.annotation.Creator;
 import io.micronaut.core.annotation.Introspected;
 import io.micronaut.core.annotation.NonNull;
 
 import javax.validation.constraints.NotBlank;
-import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
 @Introspected
 public class BookInventory {
 
     @NonNull
     @NotBlank
-    private String isbn;
+    private final String isbn;
 
-    @NonNull
-    @NotNull
-    private Integer stock;
-
-    public BookInventory() {}
+    private final int stock;
```

Comment:
Do you prefer the primitive type?  In general, I prefer to avoid primitives.

Reply:
In this case the `Integer` can't be null, so using `int` is safe (no risk of NPE) and cleaner without the annotations and validation